### PR TITLE
Create repro.md

### PR DESCRIPTION
**Pull request by mkArtakMSFT**
Created a guide about what a repro project is and linked it from issue templates.

Addresses https://github.com/dotnet/aspnetcore/issues/39605

### CODE REVIEW

File: docs/repro.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+When customers plan to report an issue with the ASP.NET Core framework, we will most likely ask them to provide a so called `minimal repro project`.
+This document describes what a minimal repro project is, and why it's important to us.
+
+## What is a minimal repro project?
+A repro, or a reproduction is a project, which can be used to reproduce the reported behavior with minimal effort from a product team, which has the minimum code required to demonstrate the concerning behavior.
+
+Here are the steps for creating such a project:
+- Create a new project, based on one of the ASP.NET Core project templates.
+- Add the minimum amount of code necessary to reproduce the behavior you try to report on this newly created project
+- Make sure you don't add any dependencies, which are irrelevant to the behavior.
```
Comment (wtgodbe) [inline docs/repro.md:12]:
```suggestion
- Make sure you don't add any dependencies that are irrelevant to the behavior.
```

The current phrasing makes it sound like you're asking them to never add any dependencies at all

**Review by wtgodbe — APPROVED** [review]
LGTM other than the 1 suggestion

File: docs/repro.md (hunk 1 above)
Comment (mkArtakMSFT) [reply docs/repro.md:12]: Thanks @wtgodbe 👍

File: docs/repro.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+A repro, or a reproduction is a project, which can be used to reproduce the reported behavior with minimal effort from a product team, which has the minimum code required to demonstrate the concerning behavior.
+
+Here are the steps for creating such a project:
+- Create a new project, based on one of the ASP.NET Core project templates.
+- Add the minimum amount of code necessary to reproduce the behavior you try to report on this newly created project
+- Make sure you don't add any dependencies that are irrelevant to the behavior.
+- Host the project as a public repository in GitHub
+- Make sure you haven't included any binaries in your project (this is usually about the `bin` and `obj` folders in your project)
+  Note: that this step is important and we won't be able to open zip attachments in your issues.
+  Those are potential attack vectors that we try to avoid at all cost.
```
Comment (dougbu) [inline docs/repro.md:16]: nit
```suggestion
  Zip files are potential attack vectors that we try to avoid at all cost.
```

**Review by dougbu — APPROVED** [review]
All nits

File: .github/ISSUE_TEMPLATE/10_bug_report.yml [hunk 3]
```diff
@@ -33,11 +33,10 @@ body:
   attributes:
     label: Steps To Reproduce
     description: |
-      Steps to reproduce the behavior.
+      Share a repro project
+      
+      We ❤ code! Point us to a [minimalistic repro project](https://github.com/dotnet/aspnetcore/blob/main/docs/repro.md) hosted in a public GitHub repo.
```
Comment (TanayParikh) [inline .github/ISSUE_TEMPLATE/10_bug_report.yml:38]:
```suggestion
      We ❤ code! Point us to a [minimalistic project which reproduces this issue (repro)](https://github.com/dotnet/aspnetcore/blob/main/docs/repro.md) hosted in a public GitHub repo.
```

File: docs/repro.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# Bug Report Reproduction Guide
+
+When customers plan to report an issue with the ASP.NET Core framework, we will most likely ask them to provide a so called `minimal repro project`.
```
Comment (TanayParikh) [inline docs/repro.md:3]:
```suggestion
When customers plan to report an issue with ASP.NET Core, we will most likely ask them to provide a so called `minimal reproduction project (repro)`.
```

File: docs/repro.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+This document describes what a minimal repro project is, and why it's important to us.
+
+## What is a minimal repro project?
+A repro, or a reproduction is a project, which can be used to reproduce the reported behavior with minimal effort from a product team, which has the minimum code required to demonstrate the concerning behavior.
+
+Here are the steps for creating such a project:
+- Create a new project, based on one of the ASP.NET Core project templates.
```
Comment (TanayParikh) [inline docs/repro.md:10]:
```suggestion
- Create a new project, based on one of the ASP.NET Core project templates.
  - **Please use the `Empty*` project templates** if they're available for that project type.
```

File: docs/repro.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+- Add the minimum amount of code necessary to reproduce the behavior you try to report on this newly created project
```
Comment (TanayParikh) [inline docs/repro.md:11]:
```suggestion
- Add the minimum amount of code necessary to reproduce the behavior you are reporting on this newly created project.
```

File: docs/repro.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+- Make sure you don't add any dependencies that are irrelevant to the behavior.
```
Comment (TanayParikh) [inline docs/repro.md:12]:
```suggestion
- Make sure you **do not** add any dependencies that are irrelevant to the behavior.
```

File: docs/repro.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+- Host the project as a public repository in GitHub
+- Make sure you haven't included any binaries in your project (this is usually about the `bin` and `obj` folders in your project)
+  Note: that this step is important and we won't be able to open zip attachments in your issues.
+  Those are potential attack vectors that we try to avoid at all cost.
+  
+  ## Important consideratoins
+  - **Never** include any sensitive information in your reproduction project
+  - **Never** include any code that is not intended to be publicized in a repro
```
Comment (TanayParikh) [inline docs/repro.md:20]:
```suggestion
  - **Never** include any sensitive information in your reproduction project.
  - **Never** include any code that is not intended to be public in a repro.
```

**Review by TanayParikh — APPROVED** [review]
Thanks! Just some nits;

I've learned recently that the "repro" term isn't necessarily known broadly. Added some comments which clarify the term in the intros.

File: docs/repro.md [hunk 9]
```diff
@@ -0,0 +1,21 @@
+# Bug Report Reproduction Guide
+
+When customers plan to report an issue with the ASP.NET Core framework, we will most likely ask them to provide a so called `minimal repro project`.
+This document describes what a minimal repro project is, and why it's important to us.
```
[the remark this answers is not in the document]
Comment (mkArtakMSFT) [reply docs/repro.md]: Thanks. I've added a note about this in the first paragraph.

File: docs/repro.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
+# Bug Report Reproduction Guide
+
+When customers plan to report an issue with ASP.NET Core, we will most likely ask them to provide a so called `minimal reproduction project (repro)`.
+This serves two purposes:
+- it helps the issue reporters to validate their assumptions by trying to recreate the behavior in a new project
+- it helps to avoid guess-work and helps with speedy investigations. Also, it helps us to provide workarounds if such applicable, much sooner.
```
Comment (TanayParikh) [inline docs/repro.md:6]:
```suggestion
- It helps issue reporters validate their assumptions by trying to recreate the behavior in a new project.
- It helps eliminate ambiguity and speeds up investigations. We may also be able to provide workarounds in certain scenarios.
```

**Pull request merged**